### PR TITLE
Add JMX Usage To User Agent String

### PR DESCRIPTION
**Pull request by sethAmazon**
# Description of the issue
We require usage metrics for jmx

# Description of changes
Add jmx usage to user agent string so it can be parsed by the backend to produce metrics. If they use jmx add it to the input. Each jmx workflow will be jmx-${workflow name} to the input. 

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Unit

Manually testing
```
User-Agent: CWAgent/1.300039.0-21-gd3672d7c-dirty-untracked (go1.22.3; linux; arm64) ID/05daabb3-ff28-4db4-ac11-377b31170b48 inputs:(jmx jmx-jvm) processors:(filter resource) outputs:(awscloudwatch cloudwatch) aws-sdk-go/1.48.6 (go1.22.3; linux; arm64)
```

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make lint`

### CODE REVIEW

File: translator/translate/otel/pipeline/jmxpipeline/translator.go [hunk 1]
```diff
@@ -4,10 +4,11 @@
 package jmxpipeline
 
 import (
-	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/pipeline/jmxresourceattributeprocessor"
 	"log"
 	"strconv"
 
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/pipeline/jmxresourceattributeprocessor"
```
Comment (sethAmazon) [inline translator/translate/otel/pipeline/jmxpipeline/translator.go:10]: did not fmt when I rebased

File: translator/translate/otel/receiver/adapter/translators.go [hunk 2]
```diff
@@ -5,10 +5,11 @@ package adapter
 
 import (
 	"fmt"
-	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/receiver/jmx"
```
Comment (sethAmazon) [inline translator/translate/otel/receiver/adapter/translators.go:8]: did not fmt when I rebased

File: translator/translate/otel/translate_otel.go [hunk 3]
```diff
@@ -6,10 +6,11 @@ package otel
 import (
 	"errors"
 	"fmt"
-	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/pipeline/jmxpipeline"
```
Comment (sethAmazon) [inline translator/translate/otel/translate_otel.go:9]: did not fmt when I rebased

File: tool/paths/paths.go [hunk 4]
```diff
@@ -10,7 +10,7 @@ const (
 	YAML           = "amazon-cloudwatch-agent.yaml"
 	ENV            = "env-config.json"
 	AGENT_LOG_FILE = "amazon-cloudwatch-agent.log"
-	JMXJarName              = "opentelemetry-jmx-metrics.jar"
+	JMXJarName     = "opentelemetry-jmx-metrics.jar"
```
Comment (sethAmazon) [inline tool/paths/paths.go:13]: did not fmt when I rebased

File: translator/translate/otel/pipeline/jmxpipeline/translator.go [hunk 5]
```diff
@@ -14,6 +13,7 @@ import (
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/common"
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/exporter/awscloudwatch"
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/extension/agenthealth"
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/pipeline/jmxresourceattributeprocessor"
```
Comment (sethAmazon) [inline translator/translate/otel/pipeline/jmxpipeline/translator.go:16]: did not fmt when I rebased

File: translator/translate/otel/receiver/adapter/translators.go [hunk 6]
```diff
@@ -73,7 +73,7 @@ var (
 	// exports to Cloudwatch while not having to follow the adapter rules
 	OtelReceivers = map[string]common.Translator[component.Config]{
 		common.OtlpKey: otlp.NewTranslator(otlp.WithDataType(component.DataTypeMetrics)),
-		common.JmxKey: jmx.NewTranslator(),
+		common.JmxKey:  jmx.NewTranslator(),
```
Comment (sethAmazon) [inline translator/translate/otel/receiver/adapter/translators.go:76]: did not fmt when I rebased

**Comment by sethAmazon** [comment]
Manually testing 
```
User-Agent: CWAgent/1.300039.0-21-gd3672d7c-dirty-untracked (go1.22.3; linux; arm64) ID/05daabb3-ff28-4db4-ac11-377b31170b48 inputs:(jmx jmx-jvm) processors:(filter resource) outputs:(awscloudwatch cloudwatch) aws-sdk-go/1.48.6 (go1.22.3; linux; arm64)
```

File: extension/agenthealth/handler/useragent/useragent_test.go [hunk 7]
```diff
@@ -138,6 +139,38 @@ func TestEmf(t *testing.T) {
 	assert.Equal(t, "outputs:(application_signals awsemf)", ua.outputsStr.Load())
 }
 
+func TestJmx(t *testing.T) {
+	metricsType, _ := component.NewType("metrics")
+	nopType, _ := component.NewType("nop")
+	jmxType, _ := component.NewType("jmx")
+	otelCfg := &otelcol.Config{
+		Service: service.Config{
+			Pipelines: map[component.ID]*pipelines.PipelineConfig{
```
Comment (sky333999) [inline extension/agenthealth/handler/useragent/useragent_test.go:148]: Can we add a test to verify the behavior when there are multiple jmx pipelines as well?

Comment (sethAmazon) [reply extension/agenthealth/handler/useragent/useragent_test.go]: Added

**Review by sky333999 — APPROVED** [review]

**Pull request merged**